### PR TITLE
Improve codebase

### PR DESCRIPTION
**Pull request by khubaib-achieve**

### CODE REVIEW

File: root/themes/modal.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Modal
-import { h5, captionMedium } from './text';
+import { h5, captionMedium } from './text.theme';
```
Comment (khubaib-achieve) [inline root/themes/modal.ts:2]: there is already `text.tsx` under `Typography`. Similarly other files in the folder can be renamed to add an extension `.theme` to make it more readable which files belong to which module.

File: root/atp-lib.tsx [hunk 2]
```diff
@@ -48,7 +66,7 @@ export default function AtpLib({
       toggleMode,
     }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [mode]
+    [mode, mergedTheme]
```
Comment (khubaib-achieve) [inline root/atp-lib.tsx:69]: if we dont add `mergedTheme` to the deps, what if `customTheme` changes, would it re-render the app?

File: root/utils/use-theme-mode.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useCallback, useEffect, useLayoutEffect, useState } from 'react';
+
+const THEME_KEY = 'theme';
+
+const useThemeMode = (usePreferences: boolean) => {
+  const [mode, setMode] = useState<Mode>(null);
+
+  const toggleMode = useCallback(() => {
+    setMode(mode === 'light' ? 'dark' : 'light');
+  }, [mode]);
```
Comment (khubaib-achieve) [inline root/utils/use-theme-mode.ts:10]: check this change

File: root/atp-lib.tsx [hunk 4]
```diff
@@ -28,15 +49,12 @@ export default function AtpLib({
...
+  const [mode, setMode, toggleMode] = useThemeMode(usePreferences);
+  const mergedTheme = useMemo(() => mergeDeep(defaultTheme, customTheme), [customTheme]);
 
   useLayoutEffect(() => {
     if (dark) {
-      if (setMode != null) {
-        setMode('dark');
-      }
-      document.documentElement.classList.add(`dark`);
+      setMode?.('dark');
```
Comment (khubaib-achieve) [inline root/atp-lib.tsx:57]: check this change

**Review by adeel-ali-atp — APPROVED** [review]

**Pull request merged**